test/infrastructure/docker/exp/api/v1alpha3/dockermachinepool_types.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha3"
+)
+
+const (
+	// MachinePoolFinalizer allows ReconcileDockerMachinePool to clean up resources
+	MachinePoolFinalizer = "dockermachinepool.infrastructure.cluster.x-k8s.io"
+)
+
+
+// DockerMachineTemplate defines the desired state of DockerMachine
+type DockerMachineTemplate struct {
+	// CustomImage allows customizing the container image that is used for
+	// running the machine
+	// +optional
+	CustomImage string `json:"customImage,omitempty"`
+
+	// PreLoadImages allows to pre-load images in a newly created machine. This can be used to
+	// speed up tests by avoiding e.g. to download CNI images on all the containers.
+	// +optional
+	PreLoadImages []string `json:"preLoadImages,omitempty"`
+
+	// ExtraMounts describes additional mount points for the node container
+	// These may be used to bind a hostPath
+	// +optional
+	ExtraMounts []infrav1.Mount `json:"extraMounts,omitempty"`
+}
+// DockerMachinePoolSpec defines the desired state of DockerMachinePool
+type DockerMachinePoolSpec struct {
+	// Template contains the details used to build a replica machine within the Machine Pool
+	// +optional
+	Template DockerMachineTemplate `json:"template"`
```

Comment:
Might be I'm missing some context here, but I find a little bit confusing having a  `DockerMachineTemplate` type different than the one defined in the main CAPD API.

Could we use `DockerMachineSpec` (similar to what we are doing for the template in `MachinePoolSpec` https://github.com/kubernetes-sigs/cluster-api/blob/cf467a0fefea032290e3261ee89d98e3d7623ca5/exp/api/v1alpha3/machinepool_types.go#L44)

Reply:
Not all of the fields in `DockerMachineTemplate` were applicable for the machine pool template. For example, `Bootstrapped` doesn't make sense on a template. `Boostrapped` makes sense on the status of an instance of a template, but not on the template itself. Also, it seems like `Bootstrapped` would be better placed on the status rather than the spec, but that's loosely related.